src/hypothesis/internal/debug.py

```diff
@@ -17,14 +17,17 @@
 
 from __future__ import division, print_function, absolute_import
 
-from hypothesis import settings as Settings
-from hypothesis.core import find
+import os
+import sys
```

Comment:
The deferred import is solely to prevent circular references between modules.

Arguably the solution here is to split this up into another module actually. I think I'll do that actually and then revert this file.